tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_graphics_Cursor.java

```diff
@@ -162,6 +164,66 @@ public void test_ConstructorWithImageDataProvider() {
 			IllegalArgumentException.class, () -> new Cursor(display, (ImageDataProvider) null, 0, 0));
 }
 
+@Test
+public void test_InvalidArgumentsForAllConstructors() {
+	ImageData source = new ImageData(16, 16, 1, new PaletteData(new RGB[] { new RGB(0, 0, 0) }));
+	ImageData mask = new ImageData(16, 16, 1, new PaletteData(new RGB[] { new RGB(0, 0, 0) }));
+
+	assertThrows("IllegalArgumentException Expected: when wrong style was provided", IllegalArgumentException.class,
```

Comment:
Parts of the error messages are rather obsolete. In particular the expectation of throwing an IllegalArgumentException is already encoded in the assertions themselves.